### PR TITLE
DRIVERS-2882 Add EKS OIDC Support

### PR DESCRIPTION
**Pull request by blink1073**
Skip aks testing for now, pending https://jira.mongodb.org/browse/BACKPORT-22539

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -853,24 +881,40 @@ tasks:
...
       commands:
-        - func: "run oidc k8s test"
+        - func: "run oidc k8s atlas test"
+          vars:
+            VARIANT: eks
+        - func: "run oidc k8s atlas test"
           vars:
             VARIANT: gke
-        - func: "run oidc k8s test"
+        # - func: "run oidc k8s atlas test"
```
Comment (NoahStapp) [inline .evergreen/config.yml:902]: Intentionally commented out?

File: .evergreen/config.yml [hunk 2]
```diff
@@ -853,24 +881,40 @@ tasks:
...
+      tags: ["latest", "oidc", "pr"]
+      commands:
+        - func: "run oidc k8s local test"
           vars:
-            VARIANT: aks
+            VARIANT: eks
+        - func: "run oidc k8s local test"
+          vars:
+            VARIANT: gke
+        # - func: "run oidc k8s local test"
```
Comment (NoahStapp) [inline .evergreen/config.yml:915]: Intentionally commented out?

File: .evergreen/k8s/aks/README.md [hunk 3]
```diff
@@ -1,7 +1,19 @@
-#
+# AKS Cluster Management
 
-## AKS Cluster Setup
+Scripts to manage a drivers test cluster on Azure.
+
+## Cluster Management
+
+These steps must be done by an admin (one time):
```
Comment (NoahStapp) [inline .evergreen/k8s/aks/README.md:7]:
```suggestion
These steps must be done by an account with admin access (one time):
```

File: .evergreen/k8s/eks/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# EKS Cluster Management
+
+Scripts to manage a drivers test cluster on AWS.
+
+## Cluster Management
+
+These steps must be done by an admin (one time):
```
Comment (NoahStapp) [inline .evergreen/k8s/eks/README.md:7]:
```suggestion
These steps must be done by an account with admin access (one time):
```

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: .evergreen/auth_oidc/k8s/remote-scripts/setup_oidc.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Set up OIDC auth roles.
+ */
+(function() {
+    "use strict";
+
+    console.log("Setting up Admin User");
+    const admin = Mongo().getDB("admin");
+    assert(admin.auth("bob", "pwd123"));
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/auth_oidc/k8s/remote-scripts/setup_oidc.js:9]: These two values are stored in multiple places throughout the repo

File: .evergreen/config.yml (hunk 1 above)
Comment (blink1073) [reply .evergreen/config.yml:902]: Yes, pending [jira.mongodb.org/browse/BACKPORT-22539](https://jira.mongodb.org/browse/BACKPORT-22539)

File: .evergreen/config.yml (hunk 2 above)
Comment (blink1073) [reply .evergreen/config.yml:915]: Yes

File: .evergreen/auth_oidc/k8s/README.md [hunk 6]
```diff
@@ -1,9 +1,111 @@
...
+The scripts can be run locally as follows:
+
+```bash
+bash setup.sh local  # needs to be done once to set up variables
+bash setup-pod.sh aks  # or gke, or eks
+bash start-server.sh
+bash run-self-test.sh
+```
+
+Or if running the driver test:
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/auth_oidc/k8s/README.md]: Updated

File: .evergreen/auth_oidc/k8s/setup-pod.sh [hunk 7]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -eu
+
+SCRIPT_DIR=$(dirname ${BASH_SOURCE[0]})
+. $SCRIPT_DIR/../../handle-paths.sh
+pushd $SCRIPT_DIR
```
[the remark this answers is not in the document]
Comment (blink1073) [reply .evergreen/auth_oidc/k8s/setup-pod.sh:6]: Fixed

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**